src/detect-lua.c

```diff
@@ -278,7 +279,14 @@ int DetectLuaMatchBuffer(DetectEngineThreadCtx *det_ctx,
                 SCLogDebug("k='%s', v='%s'", k, v);
 
                 if (strcmp(k, "retval") == 0) {
-                    if (atoi(v) == 1)
+                    int val;
+                    if (ByteExtractStringInt32(&val, 10, 0,
+                                               (const char *)v) < 0) {
+                        SCLogError(SC_ERR_INVALID_VALUE, "Invalid value in "
+                                   "the table: '%s'", v);
+                        ret = 0;
+                    }
+                    if (val == 1)
```

Comment:
That can be `else if`

Reply:
OK.

Reply:
hey @catenacyber now that I was making changes, I do not understand how this can be `else if`. Could you please explain?

Reply:
`val` can only be 1 if `ByteExtractStringInt32` succeeded

So we do not need to test this condition in the case `ByteExtractStringInt32` failed, so we can put an else